pandas/io/xml.py

```diff
@@ -330,6 +363,67 @@ def _parse_nodes(self) -> list[dict[str, str | None]]:
 
         return dicts
 
+    def _iterparse_nodes(self) -> list[dict[str, str | None]]:
+        from xml.etree.ElementTree import iterparse
+
+        dicts: list[dict[str, str | None]] = []
+        row: dict[str, str | None] | None = None
+
+        if not isinstance(self.iterparse, dict):
+            raise TypeError(
+                f"{type(self.iterparse).__name__} is not a valid type for iterparse"
+            )
+
+        row_node = next(iter(self.iterparse.keys())) if self.iterparse else ""
+        if not is_list_like(self.iterparse[row_node]):
+            raise TypeError(
+                f"{type(self.iterparse[row_node])} is not a valid type "
+                "for value in iterparse"
+            )
+
+        if (
```

Comment:
@twoertwein, your thoughts on this handling? Since `iterparse` can potentially read very large XML files (1 GB, 5 GB, 10 GB+), this checks for strings, online docs, or compressed docs. The idea is to avoid `get_handle` downloading or extracting such large content in memory and raise `MemoryError` or `OSError`

Reply:
I'm not sure, some people have lots of memory and would be fine downloading a 10GB file. Might be easier to have such a warning as part of the doc-string?

Reply:
Iterparse is memory efficient for large files. Only physical, fully extracted files on hard disk is being allowed in this method. This raises `ParserError` for online or compressed sources to avoid in-memory downloading or decompression by `get_handle`. Users may have a 10GB XML file on local disk to be iterparsed here. Do these `if` conditions exhaust all non-file system possibilities? Can `get_handle` or a related method check if path is a local file and return the path without reading content?

Reply:
Could also add a `local_only` keyword to `get_handle` but that might make get_handle even more complex.